### DISCUSSION TITLE
Generating A2L from Json & Parsing Transformers from A2L

### DISCUSSION THREAD

**Discussion by ramkrganesh** [Luncher91/A2LParser#40] [Q&A] [answered]
Hello Andreas Lenzen,

Firstly, you previous suggestions helped me getting started with parsing the A2L and thank you for that.
Using the parser, I converted the A2L file into JSON and at the moment, I have 2 questions -

1) Is there any python method out of the box which could convert the parsed JSON content into A2L ? 
    I see that there are several classes and methods in main.py but I assume they do not support generating an A2L from parsed JSON.

2) My understanding is that, this parser could not parse "Transformers" and if I have to add this feature, could you please explain how I could start with it ? Maybe about updating  java scripts ?

Best Regards,
Ram

**Comment by ramkrganesh** [comment]
Update about question 1)
I was able to generate the a2l using the following code -
```
proc = subprocess.Popen(["java", "-jar", sys.argv[1], "--json", input_file, "-o", "./output.a2l"],
                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

**Comment by Luncher91** [comment, marked as answer]
Hi Ram,

Regarding 1: I would have suggested exactly that :-D
Regarding 2: You are right, my parser does not yet support "Transformers" because it is part of the newer version of the standard which I don't have. So if you could implement the support for things from that standard I would welcome that 👍 
The parser specification has been added bei [schmo1337](https://github.com/Luncher91/A2LParser/blame/e34c328ae7975c47b3c9a6055e8ea82b434f8de9/src/main/antlr4/a2lParser.g4#L104)
So it is a matter of adding "Transformer" to the class structure (adding a new field on ModuleSubBlocks, creating class(es) for "Transformer") and adding the parsing step into the [visitor](https://github.com/Luncher91/A2LParser/blob/e34c328ae7975c47b3c9a6055e8ea82b434f8de9/src/main/java/net/alenzen/a2l/A2LVisitor.java#L210C17-L210C36)

Kind regards,
Andreas

**Reply by ramkrganesh** [reply]
Hi Andreas,

May I know which version of JAVA JDK was used to generate "a2lparser-2.6.3-3a6b8ab-jar-with-dependencies.jar" ?
I downloaded the source code and used CORRETTO-1.8 but, the output jar file seems incompatiable with pyjnius python library which I currently use.
But, the a2lparser-2.6.3.jar file is compatiable and hence the question.

Thanks !

**Reply by Luncher91** [reply]
the jars "with dependencies" include all the libraries my library depends on in a single self containing jar. The jar files without dependencies are meant to be included in another java project along with its dependencies separately.

The jar I provided has been build with the openjdk docker container maven:3-openjdk-8
Did you use maven to build the project? The `mvn clean package` executed on the root directory of the repository should bring you to your goal. I would not recommend trying to compile the project without maven since there are some generation steps for the ANTLR grammar involved.

**Reply by ramkrganesh** [reply]
Thank you for the answer.

I use IntelliJ Idea for building the project and manually imported the dependency JAR files before building.
I also installed the ANTLR plug-in and generated classess for a2lLexer, a2lParser...etc.

After adding the dependencies, I was able to build and generate output JAR file and it also worked with pyjnius.
